### DISCUSSION TITLE
"Add missing languages" does not seem to work

### DISCUSSION THREAD

**Discussion by fflorent** [WeblateOrg/weblate#11234] [Q&A] [answered]
We are using Weblate on hosted.weblate.org, in order to localize a website written in markdown (built using mkdocs).
We enabled these addons:
 - the "Component Discovery";
 - the "Add missing languages";

After having waited 24h, the languages other than English are not created despite being present in the first component (named Index) we created manually so we could enable the "Component Discovery". Do you know what could prevent that creation?

Some notes:
 1. We first have enabled the "Component Discovery" addon, then have imported a markdown file localized in French (by pulling the repository after the localized file has been created), then enabled the "Add missing languages" addon;
 2. I notice that in the component settings, the ["Add new translation" field](https://docs.weblate.org/en/latest/admin/projects.html#component-new-lang) is set to "Contact maintainers". That was the value we set initially for the Index component, I don't know if that could interfere with the "Add missing languages" addon?

PS: this question is similar to #7735, I don't know if you would have prefered starting a new discussion here or dig that thread?

**Comment by nijel** [comment, marked as answer]
Maybe adding of new languages is not configured? See https://docs.weblate.org/en/latest/devel/translations.html#adding-translation. Some formats also need https://docs.weblate.org/en/latest/admin/projects.html#component-new-base to be configured.

**Reply by fflorent** [reply]
Alright! The missing part is that we need indeed to configure the "Template for new translations" as stated here:
https://docs.weblate.org/en/latest/formats/markdown.html

If we fill this, we have our languages appearing.

However, is this something that the Component Discovery addon can automatically do itself for every new components? Or is this a manual step that has to be done?

**Reply by fflorent** [reply]
Ah, in the Component Discovery settings, is this the "Define the base file for new translation" the field needed for that?

![Define the base file for new translation filled with the same value as for "Define the monolingual base filename"](https://github.com/WeblateOrg/weblate/assets/371705/ef1ee203-bed5-4954-8a3a-6b78e9651cb6)

**Reply by nijel** [reply]
Yes.
